### PR TITLE
po: Remove all `#fuzzy,` strings from .pot files

### PR DESCRIPTION
**Pull request by marusak**
This maybe works :D I want to test with po-refresh.

 * [x] po-refresh

`xgettext` generates header, which contains line `#fuzzy,`, which is not
wanted by zanata server.

See https://github.com/zanata/zanata-server/pull/302

### CODE REVIEW

**Comment by cockpituous** [comment]
po-refresh in progress on 1-ci-srv-03.
Log: https://209.132.184.41:8493/logs/po-refresh-12703-20190904-075656/

**Comment by cockpituous** [comment]
po-refresh in progress on 3-ci-srv-04.
Log: https://209.132.184.41:8493/logs/po-refresh-12703-20190904-075656/

**Comment by cockpituous** [comment]
po-refresh  done: https://github.com/cockpituous/cockpit/commits/po-refresh-20190904-075959

**Comment by marusak** [comment]
Ok, it worked.
Not really sure why this occurred now, since the change was introduced 6 years ago and our zanata uses much newer version.

Does this change make sense to others?

**Comment by martinpitt** [comment]
Wait, what? `fuzzy` is contained in the PO *template*? This is supposed to be in *.po files, when the original string changes and thus the translation becomes invalid. It won't be used at runtime any more, but it's still useful to not completely lose the translation (often you just need to adjust a new or dropped interpunction, or a slight wording change). But it has absolutely no place in a *.pot.

Sedding out  `# fuzzy` looks like a dangerous thing, and if xgettext produces that this is a major bug there. We may need it as a temporary workaround, but it should at least be linked to a bug report. 

(Also, mis-formatted Closes ## in the commit log)

**Comment by marusak** [comment]
> Wait, what? fuzzy is contained in the PO template?

` /usr/bin/xgettext --output=- --language=JavaScript --keyword= --keyword=ngettext:1,2,3t --from-code=UTF-8 pkg/docker/containers-view.jsx`
and check 7th line.

From https://stackoverflow.com/questions/15653093/what-does-fuzzy-mean-at-the-top-of-a-xgettext-generated-pot-file
> `xgettext` adds the `#, fuzzy` entry at the top when it auto-generates a project header in the message file. The comment has no special meaning, per se: it's just noting that a human needs to come along and fill in the gaps.

There are some options to replace a few of those values like `--copyright-holder` or `--package-name` but I havn't found out how to replace all values. And `xgettext` would probably still put there the fuzzy thing.

> if xgettext produces that this is a major bug there

Seems like it worked like that since ever. I can report, but not sure if it would get accepted.

**Comment by martinpitt** [comment]
@marusak : Oh, *only* for the first meta-data branch. This is correct indeed, as it's just meta-information that isn't supposed to go into .mo files. So apparently this is a zanata bug, seems it went overzealous with this check. This still doesn't feel right, wouldn't that mean that if you call `gettext("")`, you would get back that metadata instead of the expected `""`?

**Comment by marusak** [comment]
> wouldn't that mean that if you call `gettext("")`, you would get back that metadata instead of the expected `""`?

good point, possibly it would.

There is switch `--omit-header` which would not add all of those things into `.pot` files. That would possibly make sense, since everything we add there is just template without any useful data. But possibly it is needed when pushed into zanata?
```
  # SOME DESCRIPTIVE TITLE.                                                       
  # Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER                             
  # This file is distributed under the same license as the PACKAGE package.       
  # FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.                                           
  #                                                                               
  #, fuzzy                                                                        
  msgid ""                                                                        
  msgstr ""                                                                       
  "Project-Id-Version: PACKAGE VERSION\n"                                         
  "Report-Msgid-Bugs-To: \n"                                                      
  "POT-Creation-Date: 2019-09-04 12:08+0200\n"                                    
  "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"                                     
  "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"                                  
  "Language-Team: LANGUAGE <LL@li.org>\n"                                         
  "Language: \n"                                                                  
  "MIME-Version: 1.0\n"                                                           
  "Content-Type: text/plain; charset=UTF-8\n"                                     
  "Content-Transfer-Encoding: 8bit\n"                                             
  "Plural-Forms: nplurals=INTEGER; plural=EXPRESSION;\n"         
```

**Comment by martinpitt** [comment]
@marusak: That's worth trying out. You can build it locally and run `make upload-pot` to test it. You need a valid ~/.config/zanata.ini for that and zanata-client package installed.

**Comment by martinpitt** [comment]
... or just let the infra do that again for you :)

**Comment by cockpituous** [comment]
po-refresh in progress on 1-cockpit-11.
Log: https://209.132.184.41:8493/logs/po-refresh-12703-20190904-110101/

**Comment by marusak** [comment]
hmm, so possibly (now I am letting CI run it) can omit headers, but then we cannot have utf-8 character in strings for translation (see the second commit)

**Comment by cockpituous** [comment]
po-refresh  done: https://github.com/cockpituous/cockpit/commits/po-refresh-20190904-110404

**Comment by marusak** [comment]
Update - I send email to maintainers of zanata asking some question (mainly if there were some updates that could cause this problem). I'll wait a bit if they replay.

**Comment by marusak** [comment]
> I send email to maintainers of zanata asking some question...

They said that there has been not update. So I have no idea why it stopped working now.

> there really should be some recent/open zanata bug

Zanata as a project is dead. There are plans to move to [Weblate](https://fedoraproject.org/wiki/L10N_Move_to_Weblate).

> I. e. do the *.po files still get a correct header with the encoding?

We need to just see what will be returned from zanata :)

**Comment by cockpituous** [comment]
po-refresh in progress on 2-ci-srv-05.
Log: https://209.132.184.41:8493/logs/po-refresh-12703-20190909-082525/

**Comment by cockpituous** [comment]
po-refresh  done: https://github.com/cockpituous/cockpit/commits/po-refresh-20190909-082929

**Comment by cockpituous** [comment]
po-refresh in progress on 4-ci-srv-02.
Log: https://209.132.184.41:8493/logs/po-refresh-12703-20190909-084242/

**Comment by cockpituous** [comment]
po-refresh  done: https://github.com/cockpituous/cockpit/commits/po-refresh-20190909-084646

**Comment by marusak** [comment]
Nope, this is not the right approach, see https://github.com/cockpituous/cockpit/commit/4c4c11ef99fa0c0a8fdc15a46836c1c1e575ddcd#diff-3316db9fe0395b3a4093ec5173dc1170R149

**Comment by cockpituous** [comment]
po-refresh in progress on 4-cockpit-11.
Log: https://209.132.184.41:8493/logs/po-refresh-12703-20190909-105959/

**Comment by cockpituous** [comment]
po-refresh  done: https://github.com/cockpituous/cockpit/commits/po-refresh-20190909-110101

**Comment by marusak** [comment]
So I got back to my original commit.
This works sensibly. @martinpitt your comment that this could end up with translating empty string as that header is possibly reasonable, but if you look in current translations (e.g https://github.com/cockpit-project/cockpit/blob/master/po/de.po) this `fuzzy` string is not present there anyway.

**Comment by marusak** [comment]
Added HACK and also picked the last update from the bot

File: po/de.po [hunk 1]
```diff
@@ -28,23 +28,23 @@ msgstr ""
...
 
-#: pkg/docker/storage.jsx:259
+#: pkg/docker/storage.jsx:261
 msgid " (shared with the OS)"
 msgstr " (gemeinsam mit dem Betriebssystem freigegeben)"
 
 #: pkg/storaged/others-panel.jsx:57
 msgid "$0 Block Device"
-msgstr "$0 blockorientiertes Gerät"
+msgstr "$0 blockorientiertes Ger??t"
```
Comment (martinpitt) [inline po/de.po]: This still seems to mess up translation imports, i. e. the encoding.

**Comment by cockpituous** [comment]
po-refresh in progress on 3-ci-srv-06.
Log: https://209.132.184.41:8493/logs/po-refresh-12703-20190909-124646/

**Comment by marusak** [comment]
Ok, this was walking in a wrong direction, there is different problem. I'll figure it out locally and send another PR which really fixes that.

**Pull request closed without merging**